### PR TITLE
add a utility function to regularly question a Discoverer for peers

### PR DESCRIPTION
**Pull request by marten-seemann**
The idea is the following:
1. We construct an `AutoRelay` with a `peerChan <-chan peer.AddrInfo`. Peers sent on this channel are potential candidates for v2 relays.
2. We then regularly query the DHT for new peers, and send those on the channel.

This avoids a circular dependency between the DHT (which depends on the `host.Host`), and `AutoRelay`: You can now construct your host (with `AutoRelay`) first, just passing in the `peerChan`. Once the host and the DHT are constructed, you can now start filling that chan.

I'm not sure if the implementation here makes sense though. We're regularly restarting `FindPeers`. Instead, maybe we can just continue receiving values from that chan?

### CODE REVIEW

File: p2p/discovery/util/util.go [hunk 1]
```diff
@@ -28,6 +28,45 @@ func FindPeers(ctx context.Context, d discovery.Discoverer, ns string, opts ...d
...
+			case <-t.C:
+				if cancel != nil {
+					cancel()
+				}
+				fctx, cancel = context.WithCancel(ctx)
+				var err error
+				ch, err = d.FindPeers(fctx, ns, opts...)
+				if err != nil {
+					log.Errorf("FindPeers failed: %s", err)
+					return
```
Comment (vyzo) [inline p2p/discovery/util/util.go:62]: this will short circuit in first error, probably not what we want?

**Review by vyzo — COMMENTED** [review]
we might want to adjust the query rate so that we backoff after we find some peers.

Short circuiting in the first error may be problematic here, maybe we need some more smarts? Eg check for tmp errors.

**Comment by marten-seemann** [comment]
So GitHub just closed this PR because I deleted the `discover-v2-relays` branch. And now it doesn't even allow me to change the merge target to `master`. Will have to open a new PR for this.

**Comment by marten-seemann** [comment]
https://github.com/libp2p/go-libp2p/pull/1379

**Pull request closed without merging**